pkg/goDB/engine/query.go

```diff
@@ -104,8 +128,18 @@ func (qr *QueryRunner) Run(ctx context.Context, args *query.Args) (res *results.
 		return nil, fmt.Errorf("failed to prepare query statement: %w", err)
 	}
 
-	// get list of available interfaces in the local DB
-	stmt.Ifaces, err = parseIfaceList(qr.dbPath, args.Ifaces)
+	// get list of available interfaces in the local DB, filter based on given comma separated list or regexp,
+	// reg exp is preferred
+	var dbLister = NewDBInterfaceLister(qr.dbPath)
+
+	if isIFaceArgumentRegExp(args.Ifaces) {
+		iFaceRegexpArg := args.Ifaces
+		iFacesRegExp := iFaceRegexpArg[1 : len(iFaceRegexpArg)-1]
```

Comment:
I'd really avoid using any form of string indexing (always) because you're acting on runes, not on characters. In an ideal world that doesn't matter, but the world isn't ideal... 
I'd just use one of the regex matching methods and a group to both match _and_ extract at the same time, that's probably the most elegant solution...